port/unix/omrsock.c

```diff
@@ -76,8 +76,32 @@ omrsock_getaddrinfo_create_hints(struct OMRPortLibrary *portLibrary, omrsock_add
 
 int32_t
 omrsock_getaddrinfo(struct OMRPortLibrary *portLibrary, char *node, char *service, omrsock_addrinfo_t hints, omrsock_addrinfo_t result)
-{
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+{	
+	if ((NULL == hints) || (NULL == result)) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	
+	omr_os_addrinfo *addrInfoResults = NULL;
+	omr_os_addrinfo *addrInfoHints = NULL;
+	int32_t count = 0;
```

Comment:
Variable declaration should always be placed at the start of the function before the `if` statement. Windows compilers don't allow it. If this code is used for the Windows implementation, then it won't work. You should always write code which is compiler agnostic.